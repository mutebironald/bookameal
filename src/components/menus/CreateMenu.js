import React from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import "../.././App.css"

import { addMenuRequest } from '../../actions/addMenuRequest';


class CreateMenu extends React.Component {
    constructor(props){
    super(props);
    this.state = {
        meal_id :'',
        day: ''
        }
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    }

    onChange(event) {
        this.setState({ [event.target.name]: event.target.value});
        console.log(event.target.meal_id)

    }

    componentWillReceiveProps(here){
        console.log(here)
        
    }

    onSubmit(event){
        event.preventDefault();
        if (this.state.meal_id.trim()){
            console.log(this.state);
            this.props.addMenuRequest((this.state));
            console.log(this.state)
            this.setState({meal_id:'', day: ''})
        }
    }
    render(){
        return (
            <form onSubmit={this.onSubmit}>

                <div className="form-group">
                    <label className="control-label">Meal Id</label>
                    <input
                        value={this.state.meal_id}
                        onChange={this.onChange}
                        type="number"
                        name="meal_id"
                        className="form-control"
                    />
                </div>

                <div className="form-group">
                    <label className="control-label">Day</label>
                    <input
                        value={this.state.day}
                        onChange={this.onChange}
                        type="text"
                        name="day"
                        className="form-control"
                    />
               
                    
                </div>

                <div className="form-group">
                    <button className="btn btn-primary btn-sm btn-success">
                        Create Menu
                    </button>
                </div>

            </form>
        );
    }
}
CreateMenu.propTypes = {
    addMenuRequest: PropTypes.func.isRequired
}

export default connect (null, {addMenuRequest})(CreateMenu);
