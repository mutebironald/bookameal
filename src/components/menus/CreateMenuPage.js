import React, {Component} from "react";
import CreateMenu from "./CreateMenu";
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import { addMealRequest } from '../../actions/addMealRequest';  

export class AddMenuPage extends Component{
    render(){
        const { addMealRequest } = this.props;
        return(
            <div className="row">
                <div className="col-md-4 col-md-offset-4">
                    <CreateMenu addMealRequest={addMealRequest}/>
                </div>
            </div>
        );
    }
}



AddMenuPage.propTypes = {
    addMealRequest: PropTypes.func.isRequired
}



export default connect( null , {addMealRequest}) (AddMenuPage);
