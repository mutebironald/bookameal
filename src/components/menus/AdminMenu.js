import React, { Component } from 'react';

import axios from 'axios'

import { Button, Col,
    Modal, ModalHeader, ModalBody,
    } from 'reactstrap'

import {
        Collapse,
        Navbar,
        NavbarToggler,
        NavbarBrand,
        Nav,
        NavLink
        } from 'reactstrap';


import '../.././index.css';
import '../.././App.css';

import GetMenuPage from './GetMenuPage';

import CreateMenuPage from './CreateMenuPage'

import getCurrentDay from '../current'

import WeekDays from '../Weekdays'

const selectedStyle = {
    backgroundColor: "white",
    color: "slategasy"
  }

const today = getCurrentDay();

class AdminMenu extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            menus: [],
            modal: false,
            // today: '',
            currentMenu: {
                day: "",
                meals: []
                
              }
            }
            this.toggle = this.toggle.bind(this)
            this.getMenus = this.getMenus.bind(this)
        };



    getMenus = () => {
        console.log("getMenus", axios.get(`api/v1/menu` , {headers:{"Authorization": localStorage.getItem('Authorization')}}))
        axios.get(`api/v1/menu`, {headers:{"Authorization": localStorage.getItem('Authorization')}})
        .then(response => {
            console.log("this getMenus response",today, response.data.Menu,"hererea", response.data.Menu.find(
              menu => today === menu.day
            ))
            const today = getCurrentDay();
            console.log("magicmarie",response.data.Menu.day)
            const currentMenu = response.data.Menu.find(
             menu => today.toLowerCase() === menu.day
            );
            console.log("current menu" , currentMenu)
            this.setState({
              menus: response.data.Menu,
              currentMenu
            });
            console.log("this is your state",currentMenu)
          })
        }

    getMenu = day => {
        day = day.charAt(0).toUpperCase() + day.slice(1);
        const { menus } = this.state;
        const currentMenu = menus.find(menu => day.toLowerCase() === menu.day);
        console.log("At this point debugging kicks in", currentMenu)
        this.setState({ currentDay: day.toLowerCase(), currentMenu: currentMenu });
        console.log("howday this is our current day", day, "menu", this.state)
    };


    componentDidMount() {
        this.getMenus();
      }

    

    toggle(){
        this.setState({
            modal: !this.state.modal,
        })
    }

    returnMenu = () =>{
        const  currentMenu = this.state.currentMenu;

        if(currentMenu.meals.length === undefined){
            console.log("we are here",currentMenu.meals)
            return (
                <div>
               <h2 class="text-white">No meals in menu</h2>
                <Button className="btn btn-success btn-sm" onClick={this.toggle} style={{ width: "4rem", height:"2rem"}}>Add</Button>
                <Modal isOpen={this.state.modal} backdropTransition={{ timeout: 1300 }}
                toggle={this.toggle} className={this.props.className}>
                    <ModalHeader toggle={this.toggle}>Menu</ModalHeader>
                    <ModalBody>
                        <CreateMenuPage/>      
                    </ModalBody>
                </Modal>
                </div>
            );
        }else {
           return (
               <Col>
               <div>

                <Navbar color="success" light expand="md">
                <NavbarBrand href="/">Book-A-Meal | Home </NavbarBrand>
                <NavbarToggler onClick={this.toggle}/>
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                    <NavLink href='/getMeals' activeStyle={selectedStyle}>Meals</NavLink>
                    <NavLink href='/adminMenu' activeStyle={selectedStyle}>Menu</NavLink>           
                    <NavLink href='/adminOrders' activeStyle={selectedStyle}>Orders</NavLink>
                    <NavLink href='/adminlogin' activeStyle={selectedStyle}>Logout</NavLink>
                    </Nav>      
                </Collapse>
                </Navbar>
                
                <button className="btn btn-success btn-sm" onClick={this.toggle} style={{ width: "4rem", height:"2rem",  margin: "10px"}}>Add</button>

                <div className="row">
                <div className="col-md-2">
                <div className="header text-center" style={{ margin: "5px"}}><strong><h3>Weekday</h3></strong></div>
                <ul className="list-group">
                    <WeekDays getMenu={this.getMenu} />
                </ul>
                </div>
                {/* </div> */}
                
            {currentMenu.meals.map((menu) =>
                
                <div className="card mt-2" style={{ width: "25rem", height:"5rem", margin: "10px"}} class="card mt-2 border-success text-center center">
                <div className="row text-bold">
                
                <div className="col-md-5 meal-data text-center">
                    <strong>Meal:</strong>{menu.name}
                </div>

                <div className="col-md-4 meal-data text-center">
                    <strong>Price:</strong>{menu.price}
                </div>

                {/* <div className="col-md-8 meal-data text-center">
                    <strong>Day:</strong>{menu.day}
                </div> */}
                </div>
                                   
                </div>
                
               
            )}
            <Modal isOpen={this.state.modal} backdropTransition={{ timeout: 1300 }}
            toggle={this.toggle} className={this.props.className}>
                <ModalHeader toggle={this.toggle}>Menu</ModalHeader>
                <ModalBody>
                    <CreateMenuPage/>  
                </ModalBody>
            </Modal>

            </div>
            </div>
        
            </Col>

        );
            
        }
    }
    render() {
        return (
            <div>
                {this.returnMenu()}
                
            </div>
        );
    }
}
export default AdminMenu;
