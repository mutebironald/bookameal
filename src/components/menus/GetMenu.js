import React, { Component } from 'react';

import axios from 'axios'

import { Badge, Button, Col,
    Modal, ModalHeader, ModalBody,
     ModalFooter , Form, FormGroup,
      Label, Input} from 'reactstrap'

import {
        Collapse,
        Navbar,
        NavbarToggler,
        NavbarBrand,
        Nav,
        NavItem,
        NavLink,
        NavDropdown,
        DropdownMenu,
        DropdownItem,
        UncontrolledDropdown,
        DropdownToggle,
        } from 'reactstrap';


import '../.././index.css';
import '../.././App.css';

import GetMenuPage from './GetMenuPage';

import CreateMenuPage from './CreateMenuPage'

const selectedStyle = {
    backgroundColor: "white",
    color: "slategasy"
  }

class GetMenu extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            menu: [],
            modal: false
            }
            this.toggle = this.toggle.bind(this)
        };

    componentDidMount(){
        axios.get('/api/v1/menu')
        .then(response => { console.log(response.data); return response.data })
        .then(resData => {

            let meals = resData
           this.setState({ menuMeals: meals });
           console.log("Menu-state",this.state)
           console.log("hshdgd---")
        })
    };

    toggle(){
        this.setState({
            modal: !this.state.modal,
        })
    }

    returnMenu = () =>{
        const  menu = this.state;

        if(menu.menuMeals === undefined){
            console.log(menu)
            return (
                <div>
               <h2 class="text-white">No meals in menu</h2>
                </div>
            );
        }else {
           return (
               <Col>
               <div>

                   <Navbar color="success" light expand="md">
                <NavbarBrand href="/">Book-A-Meal | Home </NavbarBrand>
                <NavbarToggler onClick={this.toggle}/>
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                    <NavLink href='/getMeals' activeStyle={selectedStyle}>Meals</NavLink>
                    <NavLink href='/getMenu' activeStyle={selectedStyle}>Menu</NavLink>           
                    <NavLink href='/getOrder' activeStyle={selectedStyle}>Orders</NavLink>
                    </Nav>      
                </Collapse>
                </Navbar>
               
            {menu.menuMeals.Menu.map((menu) =>
                
                <div className="card mt-2" style={{ width: "32rem", height:"9rem", margin: "10px"}} class="card border-success">
                <div className="row text-bold">
                <div className="col-md-9 meal-data text-center">
                    <strong>Id:</strong>{menu.id}
                </div>
                <div className="col-md-9 meal-data text-center">
                    <strong>Meal:</strong>{menu.name}
                </div>
                <div className="col-md-12 meal-data text-center">
                    <strong>Day:</strong>{menu.day}
                </div>
                </div>
                    
                </div>
               
            )}
                
            <Button className="btn btn-success btn-sm" onClick={this.toggle} style={{ width: "4rem", height:"2rem"}}>Add</Button>
            <Modal isOpen={this.state.modal} backdropTransition={{ timeout: 1300 }}
            toggle={this.toggle} className={this.props.className}>
                <ModalHeader toggle={this.toggle}>Menu</ModalHeader>
                <ModalBody>
                    <CreateMenuPage/>      
                </ModalBody>
            </Modal>

            </div>
            </Col>

        );
            
        }
    }
    render() {
        return (
            <div>
                {this.returnMenu()}
                
            </div>
        );
    }
}
export default GetMenu;
