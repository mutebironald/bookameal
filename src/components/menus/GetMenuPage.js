import React, { Component  } from "react";
import GetMenu from "./GetMenu";

class GetMenuPage extends Component{

    componentWillMount(){
        console.log("GetMenuPage")
        console.log(this.props)
    
    }
    render(){
        return(
            // <div className="card mt-2" style={{ width: "12rem", height:"8rem"}}>
            <div className="card mt-2">
                <div className="row text-muted">
                    <div className="col-md-5 meal-data">
                    <strong>Meal:</strong>{this.props.item.name}
                    </div>
                    <div className="col-md-4 meal-data">
                    <strong>Price:</strong>{this.props.item.price}
                    </div>
                    <div className="col-md-3 p-2">
                    <br/><br/>
                    {/* <button
                        className="btn btn-danger">
                        delete
                    </button> */}
                    </div>
                </div>
            </div>

        )
    }
}

export default GetMenuPage;
