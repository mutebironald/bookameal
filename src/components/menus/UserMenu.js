import React, { Component } from 'react';

import axios from 'axios'

import { 
    Modal, ModalHeader, ModalBody,
    ModalFooter 
      } from 'reactstrap'

import {
        Collapse,
        Navbar,
        NavbarToggler,
        NavbarBrand,
        Nav,
        NavLink,
       
        } from 'reactstrap';
        




import '../.././index.css';
import '../.././App.css';

import GetMenuPage from './GetMenuPage';

import CreateMenuPage from './CreateMenuPage'

import CreateOrder from '../orders/CreateOrder';

import GetOrderHistory from '../orders/GetOrderHistory';

import WeekDays from '../Weekdays'

const selectedStyle = {
    backgroundColor: "white",
    color: "slategasy"
}

class UserMenu extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            menu: [],
            modal: false,
            modal1: false,
            modal2: false,
            today: ''
            }
            this.toggle = this.toggle.bind(this)
            this.toggle1 = this.toggle1.bind(this)
            this.toggle2 = this.toggle2.bind(this)
        };


    getMenus = day => {
        axios.get(`api/v1/menu/caterer/${day}`)
        .then(response => { console.log("here -------- response", response.data.Menu);
         return response.data.Menu })
        .then(menu => {
            let meals = menu
            this.setState({ menu: meals, today: day.charAt(0).toUpperCase() + day.slice(1) });
        })
        .catch(error => console.log(error))    
    }


    getCurrentDay = () => {
        const numberDay = new Date().getDay();
        const days = [
          { key: 1, name: "Monday" },
          { key: 2, name: "Tuesday" },
          { key: 3, name: "Wednesday" },
          { key: 4, name: "Thursday" },
          { key: 5, name: "Friday" },
          { key: 6, name: "Saturday" },
          { key: 7, name: "Sunday" }
        ];
        const today = days.find(day => day.key === numberDay);
        return today.name;
      };

    getMenu = day =>{
        console.log("my state today", this.state)
        day = day.charAt(0).toUpperCase() + day.slice(1);
        const { menu } = this.state;
        this.setState({ currentDay:day , menu: [menu]});
    };

    componentDidMount(){
        
        const today = this.getCurrentDay();
        console.log("today h-----", today, typeof today, this.getMenus(today.toLowerCase()))
        this.getMenus(today.toLowerCase());
        this.setState({today});
        console.log(this.state, "componentWillMount")
    }

    toggle(){
        this.setState({
            modal: !this.state.modal
        });
    }

    resetOrder = () => this.setState({
        modal1: !this.state.modal1,
        id: null
    })

    makeOrder = () => {
        const id  = this.state.id;

        axios.post(`/api/v1/orders`, id)
        .then(response => {
            console.log("response", response.data)
            dispatchEvent({ type: "ADD_ORDER", payload: id});
        })
        .catch(error => {
            console.log("errors", error);
        })

        this.resetOrder();
    }

    toggle1(id){
        console.log(id, "my id")
        this.setState({
            modal1: !this.state.modal1,
            id: id
        });
    }

    toggle2(){
        this.setState({
            modal2: !this.state.modal2
        });
    }

    resetOrder = () =>  this.setState({
        modal1: !this.state.modal1
    });

    resetOrder2 = () =>  this.setState({
        modal2: !this.state.modal2
    });


    returnMenu = () =>{
        const  menu = this.state;

        if(menu === undefined){
            console.log(menu.menu, "line 155")
            return (
                <div>
               <h2 class="text-white">No meals in menu</h2>
                </div>
            );
        }else {
            console.log(menu.menu, "-----------treasonous banana")
           return (

            <div class="text-center" >

            <Navbar color="success" light expand="md">
                <NavbarBrand href="/">Book-A-Meal | Home </NavbarBrand>
                <NavbarToggler onClick={this.toggle}/>
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                        <NavLink href='/userMenu' activeStyle={selectedStyle}>Menu</NavLink>

                        <NavLink href="/login" activeStyle={selectedStyle}>Logout</NavLink>

        
                    </Nav>
                    
                </Collapse>
              </Navbar>

                <div>
                <div className="row">
                    <div className="col-md-2">
                    <div className="header text-center" style={{ margin: "20px"}}><strong><h3>Weekday</h3></strong></div>

                    <ul className="list-group">
                        <WeekDays getMenu={this.getMenus} />
                    </ul>
                    </div>
                    <div className="col-md-10">
                    <h4 className="header text-center">
                        {this.state.today}
                        's Menus
                    </h4>

                    <div className="col-md-20">
                <span className="btn btn-success btn-sm" onClick={() => this.toggle2()} style={{margin: "20px"}}>Order History</span>
              </div>

              <div style={{margin: "10px 10px 10px 10px"}} class="text-center">       

            {menu.menu.map(menu =>
            
                <div style={{ width: "32rem", height:"12rem", margin: "10px 10px 10px 10px"}} className="card mt-2 border-success text-center center">
                <div className="row text-bold " >
                
                <div className="col-md-9 meal-data text-center">
                    <strong>Caterer: </strong>{menu.caterer}
                </div>

                    <div className="col-md-3">
                        <span className="btn btn-success btn-sm" onClick={() => this.toggle1(menu.id)}>Order</span>
                    </div>

                    <div className="col-md-7 meal-data text-center">
                        <strong>Meal:</strong>{menu.meals.name}
                    </div>

                    <div className="col-md-7 meal-data text-center">
                        <strong>Price:</strong>{menu.meals.price}
                    </div>


                    <div className="col-md-7 meal-data text-center">
                        <strong>Day:</strong>{menu.meals.day}
                    </div>



                    
                    </div><br/><br/>
                        
                    </div>
                
                )}
        

            </div><br/><br/>

                    </div>
                    </div>
                </div>

            

            <Modal isOpen={this.state.modal1} backdropTransition={{ timeout: 1300}}
                toggle={this.toggle1} className={this.props.className}>
                <ModalHeader toggle={this.toggle1} class="modal">Make Order</ModalHeader>
                <ModalBody class="modal-main">
                    <CreateOrder id={this.state.id} toggle={this.toggle1.bind(this)}/>
                </ModalBody>
                <ModalFooter>
                    <button onClick={() => this.resetOrder()} className="btn btn-primary">Close</button>
                </ModalFooter>
                </Modal>


            <Modal isOpen={this.state.modal2} backdropTransition={{ timeout: 1300}}
                toggle={this.toggle2} className={this.props.className}>
                <ModalHeader toggle={this.toggle2} class="modal" >My Order History</ModalHeader>
                <ModalBody class="modal-main">
                    <GetOrderHistory/>
                    {/* <CreateOrder id={this.state.id} toggle={this.toggle2.bind(this)}/> */}
                </ModalBody>
                <ModalFooter>
                    {/* <button onClick={() => this.resetOrder2()} className="btn btn-primary">Close</button> */}
                </ModalFooter>
            </Modal>
 

            </div>


        );
            
        }
    }
    render() {
        return (
            // <div className="row">
            <div>
                {/* <div className="col-sm-2">{ }</div> */}
                
                <div >{this.returnMenu()}</div>
                
                
            </div>
        );
    }
}
export default UserMenu;
