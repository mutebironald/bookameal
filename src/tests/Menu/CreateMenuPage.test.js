import React from 'react';
import { AddMenuPage } from '../../components/menus/CreateMenuPage';

describe("Create menu page", () => {
    let wrapper;
    let mockFn;
    beforeEach(()  => {
        mockFn = jest.fn();
        wrapper = shallow(<AddMenuPage addMealRequest={mockFn}/>)

    })


    it("renders successfully", () => {
        expect(wrapper).toHaveLength(1);
    });

    it("renders successfully snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
})
