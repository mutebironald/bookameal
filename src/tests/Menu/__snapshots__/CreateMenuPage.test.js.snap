// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create menu page renders successfully snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddMenuPage
    addMealRequest={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-md-4 col-md-offset-4"
>
        <Connect(CreateMenu)
                addMealRequest={[Function]}
        />
</div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(CreateMenu)
          addMealRequest={[Function]}
/>,
        "className": "col-md-4 col-md-offset-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addMealRequest": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-4 col-md-offset-4"
>
          <Connect(CreateMenu)
                    addMealRequest={[Function]}
          />
</div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(CreateMenu)
            addMealRequest={[Function]}
/>,
          "className": "col-md-4 col-md-offset-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addMealRequest": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
